tools/cassandra/handler.go

```diff
@@ -47,24 +48,48 @@ type SetupSchemaConfig struct {
 func VerifyCompatibleVersion(
 	cfg config.Persistence,
 ) error {
-
-	ds, ok := cfg.DataStores[cfg.DefaultStore]
-	if ok && ds.Cassandra != nil {
-		err := CheckCompatibleVersion(*ds.Cassandra, cassandra.Version)
-		if err != nil {
+	if ds, ok := cfg.DataStores[cfg.DefaultStore]; ok {
+		if err := verifyCompatibleVersion(ds, map[string]string{
+			cassandra_db.PluginName: cassandra.Version,
+		}); err != nil {
 			return err
 		}
 	}
-	ds, ok = cfg.DataStores[cfg.VisibilityStore]
-	if ok && ds.Cassandra != nil {
-		err := CheckCompatibleVersion(*ds.Cassandra, cassandra.VisibilityVersion)
-		if err != nil {
+
+	if ds, ok := cfg.DataStores[cfg.VisibilityStore]; ok {
+		if err := verifyCompatibleVersion(ds, map[string]string{
+			cassandra_db.PluginName: cassandra.VisibilityVersion,
+		}); err != nil {
 			return err
 		}
 	}
+
 	return nil
 }
 
+func verifyCompatibleVersion(
+	ds config.DataStore,
+	expectedVersions map[string]string,
+) error {
+	var expectedVersion string
+	var cfg config.NoSQL
+	if ds.NoSQL != nil {
+		var ok bool
+		if expectedVersion, ok = expectedVersions[ds.NoSQL.PluginName]; !ok {
+			return fmt.Errorf("unknown NoSQL plugin name: %v", ds.NoSQL.PluginName)
+		}
+		cfg = *ds.NoSQL
+	} else if ds.Cassandra != nil {
```

Comment:
You can remove this check I believe. We are always refilling NoSQL from Cassandra.

Reply:
I think we need to support old config files that may still be used by some open source customers and in our internal repo?

Reply:
Yes, but it’s back filled here so NoSQL struct should not be nil https://github.com/uber/cadence/blob/723ecf572b2391ad3acc93c455445d9968170eef/common/config/persistence.go#L53

Reply:
Ah, don't know we have the backfill logic there. Thanks for pointing that out. Removed the check on `ds.Cassandra` field.